advanced_server.rb

```diff
@@ -0,0 +1,314 @@
+        case @payload['action']
+        when 'created'
+          initiate_check_run
+        when 'rerequested'
+          create_check_run
+        when 'requested_action'
+          take_requested_action
+        end
+      end
+    end
+  end
+
+
+  helpers do
+
+    # Create a new check run
+    def create_check_run
+      # Octokit doesn't yet support the checks API, but it does provide generic HTTP methods we can use!
+      # https://developer.github.com/v3/checks/runs/#create-a-check-run
+      check_run = @installation_client.post("repos/#{@payload['repository']['full_name']}/check-runs", {
+          accept: 'application/vnd.github.antiope-preview+json', # This header allows for beta access to Checks API
+          name: 'Octo Rubocop',
+          # The information we need should probably be pulled from persistent storage, but we can
+          # use the event that triggered the run creation. However, the structure differs depending on whether
+          # it was a check run or a check suite event that trigged this call.
+          head_sha: @payload['check_run'].nil? ? @payload['check_suite']['head_sha'] : @payload['check_run']['head_sha']
+      })
+
+      # We've now requested the creation of a check run from GitHub. We will wait until we get a confirmation
+      # from GitHub, and then kick off our CI process from there.
+    end
+
+    # Start the CI process
+    def initiate_check_run
+      # This method is called in response to GitHub acknowledging our request to create a check run.
+      # We'll first update the check run to "in progress"
+      # Then we'll run our CI process
+      # Then we'll update the check run to "completed" with the CI results.
+
+      # Octokit doesn't yet support the Checks API, but it does provide generic HTTP methods we can use!
+      # https://developer.github.com/v3/checks/runs/#update-a-check-run
+      # notice the verb! PATCH!
+      updated_check_run = @installation_client.patch("repos/#{@payload['repository']['full_name']}/check-runs/#{@payload['check_run']['id']}", {
+          accept: 'application/vnd.github.antiope-preview+json', # This header is necessary for beta access to Checks API
+          name: 'Octo Rubocop',
+          status: 'in_progress',
+          started_at: Time.now.utc.iso8601
+      })
+
+      # ***** RUN A CI TEST *****
+      # This is where we would kick off our CI process. Ideally this would be performed async, so we could
+      # return immediately. But for now we'll do a simulated CI process syncronously, and update the check run right here..
+
+      full_repo_name = @payload['repository']['full_name']
+      repository     = @payload['repository']['name']
+      head_sha       = @payload['check_run']['head_sha']
+      repository_url = @payload['repository']['html_url']
+
+      clone_repository(full_repo_name, repository, head_sha)
+
+      @report = `rubocop '#{repository}/*' --format json` # --auto-correct`
+      `rm -rf #{repository}`
```

Comment:
To be fair, I didn't do any better 😹 
https://github.com/DEGoodmanWilson/excellent-ci

Reply:
@DEGoodmanWilson 🙈 I didn't see that code until just now. Our ecosystem docs issue was referencing this repo: `DEGoodmanWilson/checks-api-ruby-demo`. Is looks like simplabs/excellent is much easier to use, and we could use the GitHub Git API with it to avoid command line completely. It looks like the only advantage of using to Rubcop versus excellent is that Rubocop can demonstrate the Checks API actions feature because it supports `--auto-correct`.

Reply:
RuboCop is better-known, and I _love_ the addition of the actions feature. Let's go with what you have.